src/main/java/steps/BaseSteps.java

```diff
@@ -54,4 +55,31 @@ public static void goToCart() {
     public static void validateURL(String url) {
         Assert.assertEquals("Guest Checkout URL is not valid", url, getWebDriverInstance().getCurrentUrl());
     }
+
+    public static void validateNewTabURL(String expectedUrl) {
```

Comment:
I think that changing the pages should be part of the engine (for example WebDriverFactory or something).

Reply:
And after implementing interaction with different tabs, after the tab is switched you can simply use `validateURL` method to validate the URL of the currently opened tab in the test :)

Reply:
@marikacode is there a progress in implementing this in a way that Michal suggested (e.g. within the `WebDriverFactory` class)?